franklin-crypto/src/circuit/rescue.rs

```diff
@@ -0,0 +1,271 @@
+extern crate num_bigint;
+use self::num_bigint::{BigUint,ToBigUint};
+use std::str::FromStr;
+use super::uint32::UInt32;
+use super::multieq::MultiEq;
+use super::boolean::Boolean;
+use bellman::{ConstraintSystem, SynthesisError};
+use bellman::pairing::Engine;
+use circuit::Assignment;
+use bellman::pairing::ff::{
+    Field,
+    SqrtField,
+    PrimeField,
+    PrimeFieldRepr,
+    BitIterator
+};
+use bellman::{
+    LinearCombination,
+    Variable
+};
+use super::num::AllocatedNum;
+
+
+pub fn generate_powers<CS,E>(
+    mut cs: CS,
+    x: &AllocatedNum<E>,
+    alpha: &E::Fr
+) -> Result<AllocatedNum<E>, SynthesisError>
+    where CS: ConstraintSystem<E>,E:Engine
+{
+        assert_ne!(*alpha,E::Fr::zero());
+        assert_ne!(*alpha,E::Fr::one());
+        let alpha_bits_be = {
+            let mut tmp = Vec::with_capacity(E::Fr::NUM_BITS as usize);
+            let mut found_one = false;
+            for b in BitIterator::new(alpha.into_repr()) {
+                found_one |= b;
+                if !found_one {
+                    continue;
+                }
+                tmp.push(b);
+            }
+            tmp
+        };
+        let squares={//define square, square of square, etc variables
+            let mut tmp:Vec<AllocatedNum<E>>=vec![x.clone()];
+            //alpha is constant
+            for i in 1..alpha_bits_be.len(){
+                let sqr=tmp.last().unwrap().square(cs.namespace(|| format!("sqr{}",i)))?;
+                tmp.push(sqr);
+            }
+            tmp
+        };
+        assert_eq!(squares.len(),alpha_bits_be.len());
+        let res={
+            let n=alpha_bits_be.len();
+            let mut tmp=squares[n-1].clone();
+            //alpha is constant
+            for i in 1..n{
+                if alpha_bits_be[i] {
+                    tmp=squares[n-1-i].mul(cs.namespace(|| format!("mul_due_to_bit{}",i)),&tmp)?;
+                }
+            }
+            tmp.clone()
+        };
+        Ok(res) 
+}
+fn generate_roots<CS,E>(
+    mut cs: CS,
+    x: &AllocatedNum<E>,
+    alphas: &(Option<E::Fr>,Option<E::Fr>)
+) -> Result<AllocatedNum<E>, SynthesisError>
+    where CS: ConstraintSystem<E>,E:Engine
+{
+        //Calculate the root
+        let result_var=AllocatedNum::alloc(cs.namespace(|| "found root"), ||{
```

Comment:
please use meaningful names, e.g. `result_var` => `root`.

regarding the `_var` suffix, please be consistent and either use it for all vars or for none.